packages/run-protocol/src/runStake/runStakeManager.js

```diff
@@ -5,123 +5,156 @@ import { makeRatio } from '@agoric/zoe/src/contractSupport/ratio.js';
 import { fit, getCopyBagEntries, M } from '@agoric/store';
 import { makeNotifierKit, observeNotifier } from '@agoric/notifier';
 import { E } from '@endo/far';
+import { defineKind } from '@agoric/vat-data';
 import { makeTracer } from '../makeTracer.js';
 import { chargeInterest } from '../interest.js';
 import { KW } from './runStakeKit.js';
+import { checkDebtLimit } from '../contractSupport.js';
 
-const { details: X, quote: q } = assert;
+const { details: X } = assert;
 
-const trace = makeTracer('RM'); // TODO: how to turn this off?
+const trace = makeTracer('RSM', false);
 
 /**
- * @param {ZCF} zcf
- * @param {ZCFMint<'nat'>} debtMint
- * @param {{ debt: Brand<'nat'>, Attestation: Brand<'copyBag'>, Stake: Brand<'nat'> }} brands
- * @param {{
+ * @typedef {{
  *  getDebtLimit: () => Amount<'nat'>,
  *  getInterestRate: () => Ratio,
  *  getMintingRatio: () => Ratio,
  *  getLoanFee: () => Ratio,
- * }} paramManager
- * @param {MintAndReallocate} mintAndReallocateWithFee
- * @param {BurnDebt} burnDebt
+ * }} ParamManager
+ * @typedef {{
+ * compoundedInterest: Ratio,
+ * latestInterestUpdate: NatValue,
+ * totalDebt: Amount<'nat'>,
+ * }} AssetState
```

Comment:
reading these without indentation is awkward. Please let's indent, even though we don't yet have tool support.